icu4c/source/common/rbbi_cache.cpp

```diff
@@ -207,6 +213,7 @@ void RuleBasedBreakIterator::DictionaryCache::populateDictionary(int32_t startPo
 RuleBasedBreakIterator::BreakCache::BreakCache(RuleBasedBreakIterator *bi, UErrorCode &status) :
         fBI(bi), fSideBuffer(status) {
     reset();
+    fCjkEngine = fBI->getLanguageBreakEngine(0x3041);
```

Comment:
so, before your PR, if the user only use RuleBasedBreakIterator to break non CJK text, the CjBreakEngine will never be construct because there are no CJ text calling into RuleBasedBreakIterator and therefore we will never callfBI->getLanguageBreakEngine() with a unicode which in that CJ text range. But with your change, now you ALWAYS force to create a CjBreakEngine here , and therefore ALWAYS load megabytes of cj dictionaries into memory even if your caller is just try to break "Hello Word". This mean all the western language users now will load the cj dictionary and create the CjBreakEngine even they never come across any cj text. I do not think this practice is acceptable. It will slow down the performance for not target user as well as increase memory usage on user do not use CJ text.

Reply:
Resolved